### PR TITLE
BufferAttribute: honor itemSize in applyMatrix3()

### PR DESCRIPTION
**Pull request by dumaron**
Relative to issue #19567.

In the current version, the method `applyMatrix3` of `BufferAttribute` results in `NaN` in the attribute's array since it presumes that `itemSize` is 3. With this commit, `applyMatrix3` perform an initial check and so behaves correctly if `itemSize` is 2.

First PR here, I hope everything is fine :) .

### CODE REVIEW

File: src/core/BufferAttribute.js [hunk 1]
```diff
@@ -199,15 +200,32 @@ Object.assign( BufferAttribute.prototype, {
...
+				_vector2.y = this.getY( i );
 
-			this.setXYZ( i, _vector.x, _vector.y, _vector.z );
+				_vector2.applyMatrix3( m );
+
+				this.setXY( i, _vector2.x, _vector2.y, );
+
+			}
+
+		} else {
```
Comment (WestLangley) [inline src/core/BufferAttribute.js:216]: I would suggest this, instead... I think.
```js
} else if ( this.itemSize === 3 ) {
```

File: src/core/BufferAttribute.js [hunk 2]
```diff
@@ -199,15 +200,32 @@ Object.assign( BufferAttribute.prototype, {
...
+		if ( this.itemSize === 2 ) {
 
-			_vector.x = this.getX( i );
-			_vector.y = this.getY( i );
-			_vector.z = this.getZ( i );
+			for ( let i = 0, l = this.count; i < l; i ++ ) {
 
-			_vector.applyMatrix3( m );
+				_vector2.x = this.getX( i );
+				_vector2.y = this.getY( i );
```
Comment (WestLangley) [inline src/core/BufferAttribute.js:208]: Maybe this is better than what we were doing...
```js
_vector2.fromBufferAttribute( this, i );
```

Comment (Mugen87) [reply src/core/BufferAttribute.js:208]: Indeed 👍 . Same for `_vector`.

Comment (dumaron) [reply src/core/BufferAttribute.js]: I didn't know this method, it seems better.

File: src/core/BufferAttribute.js (hunk 1 above)
Comment (dumaron) [reply src/core/BufferAttribute.js]: Ok. Just a question: what should three do if `itemSize` differs from 2 and 3? Is this case possible?

**Review by Mugen87 — APPROVED** [review]

File: src/core/BufferAttribute.js (hunk 1 above)
Comment (WestLangley) [reply src/core/BufferAttribute.js]: >what should three do if itemSize differs from 2 and 3?

Nothing for now.

>Is this case possible?

Yes, but a use case has not been presented.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**